node/cli/src/lib.rs

```diff
@@ -120,37 +177,21 @@ pub fn start_service(
         beacon_block_rx
     );
 
-    // Create protocol and the network_task.
     // Note, that network and RPC are using the same channels to send transactions and receipts for
     // processing.
-    let (net_messages_tx, net_messages_rx) = channel(1024);
-    let protocol_config = ProtocolConfig::default();
-    let protocol = Protocol::<_, SignedBeaconBlockHeader>::new(
-        protocol_config,
+    let (messages_handler_task, network_task) = get_network_tasks(
+        p2p_port,
+        test_node_index,
         beacon_chain.clone(),
         beacon_block_tx.clone(),
         transactions_tx.clone(),
-        receipts_tx.clone(),
-        net_messages_tx.clone()
-    );
-    let network_service = Arc::new(Mutex::new(new_network_service(
-        &protocol_config,
-        NetworkConfiguration::default(),
-    )));
-    let (network_task, messages_handler_task) = create_network_task(
-        network_service,
-        protocol,
-        net_messages_rx,
     );
     let consensus_task = consensus_task_fn(
         transactions_rx,
         &beacon_block_consensus_body_tx,
     );
     tokio::run(future::lazy(|| {
-        tokio::spawn(future::lazy(|| {
-            server.wait();
-            Ok(())
-        }));
+        tokio::spawn(rpc_server_task);
```

Comment:
We are starting to need a struct that would represent a config. Also we might want to put arg-parsing stuff in a separate file.

Reply:
can make a struct for config..

separating arg parsing stuff was the point of having cli separate from servicei'm inclined to defer until we do that.